Weapon

amandaperrine opened the pull request:
Added in weapons and updated the necessary files. Closes issue #16

amandaperrine committed:
Create nb-configuration.xml

amandaperrine committed:
Update pom.xml

amandaperrine committed:
Create AssaultRifle.java

amandaperrine committed:
Create Axe.java

amandaperrine committed:
Create Crowbar.java

amandaperrine committed:
Create FryingPan.java

amandaperrine committed:
Create Pistol.java

amandaperrine committed:
Create Shotgun.java

amandaperrine committed:
Create SubmachineGun.java

amandaperrine committed:
Create Weapon.java

amandaperrine committed:
Update Fatality.java

amandaperrine committed:
Update Character.java

amandaperrine committed:
Update Survivor.java

amandaperrine committed:
Update Tank.java

amandaperrine committed:
Update GameControl.java

amandaperrine committed:
Update Party.java

amandaperrine committed:
Update SurvivorParty.java

amandaperrine committed:
Update ZombieParty.java

amandaperrine committed:
Update GameView.java

amandaperrine committed:
Update Survivor.java

JordanD-CSP commented on ZombieGame/src/main/java/AssaultRifle.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Axe.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Crowbar.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/FryingPan.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Pistol.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Shotgun.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/SubmachineGun.java:
```diff
@@ -0,0 +1,20 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Weapon.java:
```diff
@@ -0,0 +1,44 @@
+
```
`package zombiegame.app.weapon;`

JordanD-CSP commented on ZombieGame/src/main/java/Weapon.java:
```diff
@@ -0,0 +1,44 @@
+        this.name = name;
+    }
+
+    public int power() {
```
`public int useWeapon() {`

This would match it to the interface we agreed on.

JordanD-CSP commented on ZombieGame/src/main/java/Weapon.java:
```diff
@@ -0,0 +1,44 @@
+        }
+        return damage;
+    }
+
```
`public String getName() {
    return this.name;
}`

Also specified in the design.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/battle/Fatality.java:
```diff
@@ -1,41 +1,63 @@
     private String attacker; // Name of Character that attacked
     private String casualty; // Name of Character killed by attacker
-    
+    private String extraInfor;
```
`private String weapon;`

This is more descriptive and part of the design doc.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/battle/Fatality.java:
```diff
@@ -1,41 +1,63 @@
         this.casualty = casualty;
     }
+
+    public String getExtraInfor() {
+        return extraInfor;
+    }
+
+    public void setExtraInfor(String extraInfor) {
+        this.extraInfor = extraInfor;
+    }
```
`public String getWeapon() {
    return weapon;
}

public void setWeapon(String weapon) {
    this.weapon = weapon;
}`

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/battle/Fatality.java:
```diff
@@ -1,41 +1,63 @@
     public Fatality(String attacker, String casualty) {
         this.attacker = attacker;
         this.casualty = casualty;
+        this.extraInfor = null;
```
`this.weapon = null;`

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Character.java:
```diff
@@ -17,7 +17,7 @@
 public class Character {
     // Attributes
-    private String name;
+    String name;
```
`private String name;`

We should leave this as private. `name` already has getter and setter methods.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Survivor.java:
```diff
@@ -8,18 +11,46 @@
+
+    private Weapon weapon;
+    // there are 7 types of weapon
+    private static Weapon[] weapons = {
+        new Shotgun(), new SubmachineGun(),
+        new AssaultRifle(), new Axe(), new Crowbar(),
+        new FryingPan(), new Pistol()
+    };
```
This is info is meant to be included in the Equippable interface.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Survivor.java:
```diff
@@ -8,18 +11,46 @@
 
     public Survivor(String name, int health, int attack) {
         super(name, health, attack);
+        weapon = weapons[(int) (Math.random() * weapons.length)];
```
This should be handled by Equippable.equip().

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Survivor.java:
```diff
@@ -8,18 +11,46 @@
-    
-}
 
+    /**
+     * getter
+     *
+     * @return
+     */
+    public Weapon getWeapon() {
+        return weapon;
+    }
```
This should override Equippable.getWeapon().

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Survivor.java:
```diff
@@ -8,18 +11,46 @@
+        weapon = weapons[(int) (Math.random() * weapons.length)];
+    }
+
+    /**
+     * survivor could use weapon
+     *
+     * @return
+     */
+    @Override
+    int getAttack() {
+        return super.getAttack() + weapon.power();
     }
```
This logic should be handled by Party.attack(). It would be in our interest to have the getAttack() method work uniformly across all descendants of Character. That will make our code easier to maintain.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/character/Tank.java:
```diff
@@ -1,25 +1,29 @@
-    };
-    
+    }
+;
```
We can probably delete this semicolon.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/party/Party.java:
```diff
@@ -43,19 +51,26 @@ public void attack(Party defendingParty, BattleInfo battleInfo) {
                         if (defender.isDead()) {
                             attackerName = attacker.getName();
                             casualtyName = defender.getName();
-                            battleInfo.add(new Fatality(attackerName, casualtyName));
+                            Fatality fatality
+                                    = new Fatality(attackerName, casualtyName);
+                            battleInfo.add(fatality);
+                            if (attacker instanceof Survivor) {
+                                Survivor survivor = (Survivor) attacker;
+                                fatality.setExtraInfor(
+                                        survivor.getWeapon().toString());
+                            }
```
This code would need to be revised given the suggested changes. It is also part of separate task, so I think you should let someone else handle it so that we all get a chance to contribute.

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/app/battle/Fatality.java:
```diff
@@ -1,41 +1,63 @@
+
+>>>>>>> Stashed changes:ZombieGame/src/main/java/Fatality.java
     private String attacker; // Name of Character that attacked
     private String casualty; // Name of Character killed by attacker
-    
+    private String extraInfor;
+
     public Fatality(String attacker, String casualty) {
         this.attacker = attacker;
         this.casualty = casualty;
+        this.extraInfor = null;
     }
-    
+
     public String getAttacker() {
         return this.attacker;
     }
+
     public void setAttacker(String attacker) {
         this.attacker = attacker;
     }
-    
+
     public String getCasualty() {
         return this.casualty;
     }
+
     public void setCasualty(String casualty) {
         this.casualty = casualty;
     }
+
+    public String getExtraInfor() {
+        return extraInfor;
+    }
+
+    public void setExtraInfor(String extraInfor) {
+        this.extraInfor = extraInfor;
+    }
+
 }
```
This should be public given that it will be used outside this package. Can you discard these changes? They are part of a separate task and everyone should get an opportunity to contribute

JordanD-CSP commented on ZombieGame/src/main/java/zombiegame/ui/GameView.java:
```diff
@@ -8,84 +8,96 @@
             String attacker = fatality.getAttacker();
             String casualty = fatality.getCasualty();
-            System.out.printf("%s killed %s\n", attacker, casualty);
+
+            // 
+            String extraInfor = fatality.getExtraInfor();
+            if (extraInfor == null) {
+                // zombie case
+                System.out.printf("%s killed %s\n", attacker, casualty);
+            } else {
+                // survivor case
+                System.out.printf("%s killed %s with \n\t %s\n",
+                        attacker, casualty, extraInfor);
+            }
+
```
This would need to be revised given the suggested changes. It is also part of a separate task, and we should give everyone the opportunity to contribute.

JordanD-CSP committed:
Update AssaultRifle.java

- added class to package zombiegame.app.weapon

JordanD-CSP committed:
Update Axe.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update Crowbar.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update FryingPan.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update Pistol.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update Shotgun.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update SubmachineGun.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update Weapon.java

- added to package zombiegame.app.weapon

JordanD-CSP committed:
Update Weapon.java

- updated power() to useWeapon() to match to interface

JordanD-CSP committed:
Update Weapon.java

- added getName()

JordanD-CSP committed:
Update Fatality.java

- resolved conflicts
- reverted changes to match /main
- these changes were out of scope for this task

JordanD-CSP committed:
Update Character.java

- update this.name to private

JordanD-CSP committed:
Update Survivor.java

- matching to current file in /main
- changes were out of scope for task

JordanD-CSP committed:
Update Tank.java

- matched to current file in /main
- retained stylistic changes

JordanD-CSP committed:
Update Party.java

- matched to existing file in /main
- changes were out of scope for task
- retained minor stylistic changes

JordanD-CSP committed:
Update ZombieParty.java

- matched to existing file in /main
- retained minor stylistic changes

JordanD-CSP committed:
Update GameView.java

- matched to existing file in /main
- changes were out of scope for task

JordanD-CSP committed:
Merge branch 'main' into Weapon

JordanD-CSP committed:
Update SurvivorParty.java

- matched to existing file in /main
- kept minor stylistic changes

JordanD-CSP committed:
Update Weapon classes

- moved all Weapon classes into package zombiegame.app.weapon
